lua/xcodebuild/helpers.lua

```diff
@@ -95,6 +95,32 @@ function M.get_major_os_version()
   return settings.os and tonumber(vim.split(settings.os, ".", { plain = true })[1]) or nil
 end
 
+---Wraps any nvim_buf_set_option() call to ensure forward compatibility.
+---The function is required because nvim_buf_set_option() was deprecated in nvim-0.10.
+---@param bufnr number
+---@param name string
+---@param value any
+function M.set_buf_option(bufnr, name, value)
+  if vim.fn.has("nvim-0.10") == 1 then
+    vim.api.nvim_set_option_value(name, value, { buf = bufnr })
+  else
+    vim.api.nvim_buf_set_option(bufnr, name, value)
+  end
+end
+
+---Wraps any nvim_win_set_option() call to ensure forward compatibility.
+---The function is required because nvim_win_set_option() was deprecated in nvim-0.10.
+---@param winnr number
+---@param name string
+---@param value any
+function M.set_win_option(winnr, name, value)
```

Comment:
Functions are incorrectly named, you use in code `buf_set_option` and `win_set_option`. I would rename these to match the calls in the code.

Reply:
Weird that LSP didn't detect this issue